java/ql/src/experimental/Security/CWE/CWE-117/LogInjection.help

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+
+<p>If unsanitized user input is written to a log entry, a malicious user may be able to forge new log entries.</p>
+
+<p>Forgery can occur if a user provides some input with characters that are interpreted
+when the log output is displayed. If the log is displayed as a plain text file, then new
+line characters can be used by a malicious user. If the log is displayed as HTML, then
+arbitrary HTML may be included to spoof log entries.</p>
+</overview>
+
+<recommendation>
+<p>
+User input should be suitably sanitized before it is logged.
+</p>
+<p>
+If the log entries are plain text then line breaks should be removed from user input, using for example
+<code>String replace(char oldChar, char newChar)</code> or similar. Care should also be taken that user input is clearly marked
+in log entries, and that a malicious user cannot cause confusion in other ways.
+</p>
+<p>
+For log entries that will be displayed in HTML, user input should be HTML encoded before being logged, to prevent forgery and
+other forms of HTML injection.
+</p>
+
+</recommendation>
+
+<example>
+<p>In the example, a username, provided by the user, is logged using `logger.warn` (from `org.slf4j.Logger`). 
+In the first case (`/bad` endpoint), the username is logged without any sanitization.
+If a malicious user provides `Guest'%0AUser:'Admin` as a username parameter, 
+the log entry will be splitted in two separate lines, where the first line will be `User:'Guest'`, while the second one will be `User:'Admin'`.
+</p>
+<p> In the second case (`/good` endpoint), <code>replace</code> is used to ensure no line endings are present in the user input.
+If a malicious user provides `Guest'%0AUser:'Admin` as a username parameter, 
+the log entry will not be splitted in two separate lines, resulting in a single line `User:'Guest'User:'Admin'`.
+</p>
+
+<sample src="examples/LogInjection.java" />
+</example>
+
+<references>
+<li>OWASP: <a href="https://www.owasp.org/index.php/Log_Injection">Log Injection</a>.</li>
```

Comment:
Recommend linking straight to the new URL (https://owasp.org/www-community/attacks/Log_Injection) -- perhaps update other links in other qhelp files at the same time.